src/wasi.ts

```diff
@@ -826,8 +827,21 @@ export default class WASI {
       sched_yield() {},
       random_get(buf: number, buf_len: number) {
         const buffer8 = new Uint8Array(self.inst.exports.memory.buffer);
-        for (let i = 0; i < buf_len; i++) {
-          buffer8[buf + i] = (Math.random() * 256) | 0;
+        const end = buf + buf_len;
+
+        if (self.hasCrypto) {
```

Comment:
Can you check `"crypto" in globalThis` instead and not cache it in an instance variable?

Reply:
I tried, but it didn't pass the test. Looks like `globalThis` doesn't exist on all environments :disappointed:

Reply:
Also, caching it is most probably faster than checking every time `random_get` gets called

Reply:
Did it fail in node.js? Maybe the node.js version that was used is too old? `globalThis` is supposed to work in all environments. It was introduced as way to unify `window`, `self` and `global` across the various javascript environments.

As for performance, if `random_get` is called rarely a tiny slowdown shouldn't matter and if it is called frequently, I would expect the JIT to optimize it down to a couple instructions max, which is much less than even the overhead of making a native call, let alone actually running the PRNG.

Reply:
I remembered it failing on some test, but it seems I was mistaken. I've pushed the uncached `"crypto" in globalThis` version.